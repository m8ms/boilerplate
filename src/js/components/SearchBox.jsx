import React from 'react';
import { withRouter } from 'react-router';

/**
 * todo comment
 */
class SearchBox extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            query: this.props.match.params.query || ''
        }
    }

    static getDerivedStateFromProps(nextProps) {
        return {
            query: nextProps.match.params.query || ''
        };
    }

    handleSubmit = (e) => {
        e.preventDefault();

        if (this.state.query && this.state.query.length > 1) {
            this.props.history.push('/search/' + this.state.query);
        }
    };

    handleChange = (e) => {
        this.setState({query: e.target.value});
    };

    render() {

        return (
            <form className="search-box" onSubmit={this.handleSubmit}>
                <input type="text"
                       className="search-box__input"
                       placeholder="search"
                       onChange={this.handleChange}
                       value={this.state.query}/>
                <button className="search-box__submit"/>
            </form>
        )
    }
}

export default withRouter(SearchBox);
