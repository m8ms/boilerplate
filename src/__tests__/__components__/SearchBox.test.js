import React from 'react';
import SearchBox from '../../js/components/SearchBox';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

it('renders correctly with no query', () => {
    const tree = renderer
        .create(<Router><SearchBox/></Router>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
