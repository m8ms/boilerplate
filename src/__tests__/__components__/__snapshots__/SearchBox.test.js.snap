// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with no query 1`] = `
<form
  className="search-box"
  onSubmit={[Function]}
>
  <input
    className="search-box__input"
    onChange={[Function]}
    placeholder="search"
    type="text"
    value=""
  />
  <button
    className="search-box__submit"
  />
</form>
`;
